Source/WebCore/html/HTMLAttachmentElement.cpp

```diff
@@ -71,14 +76,80 @@ HTMLAttachmentElement::~HTMLAttachmentElement() = default;
 
 Ref<HTMLAttachmentElement> HTMLAttachmentElement::create(const QualifiedName& tagName, Document& document)
 {
-    return adoptRef(*new HTMLAttachmentElement(tagName, document));
+    Ref attachment = adoptRef(*new HTMLAttachmentElement(tagName, document));
+    if (document.settings().attachmentWideLayoutEnabled() && DeprecatedGlobalSettings::attachmentElementEnabled()) {
+        ASSERT(attachment->m_implementation == Implementation::Legacy);
+        ASSERT(!attachment->renderer()); // Switch to modern style *must* be done before renderer is created!
+        attachment->m_implementation = Implementation::Modern;
+        attachment->ensureUserAgentShadowRoot();
+    }
+    return attachment;
+}
+
+void HTMLAttachmentElement::didAddUserAgentShadowRoot(ShadowRoot& root)
+{
+    if (m_implementation == Implementation::Modern)
+        ensureModernShadowTree(root);
+}
+
+void HTMLAttachmentElement::ensureModernShadowTree(ShadowRoot& root)
+{
+    ASSERT(m_implementation == Implementation::Modern);
+    if (m_elementWithTitle)
+        return;
+
+    static MainThreadNeverDestroyed<const String> shadowStyle(StringImpl::createWithoutCopying(attachmentElementShadowUserAgentStyleSheet, sizeof(attachmentElementShadowUserAgentStyleSheet)));
+    auto style = HTMLStyleElement::create(HTMLNames::styleTag, document(), false);
+    style->setTextContent(String { shadowStyle });
+    root.appendChild(WTFMove(style));
+
+    auto container = HTMLDivElement::create(document());
+    static MainThreadNeverDestroyed<const AtomString> containerId("attachment-container"_s);
```

Comment:
It might be nice to put these ids in their own method, similar to what's done in `ImageControlsMac`.

Reply:
It seemed heavy, since I onlyd use the ids once. In `ImageControlsMac` they're used twice.
But it could be useful in the future, and it's more consistent between similar files, so I'll make the change.